cloudsql/sqlserver/database-sql/cloudsql.go

```diff
@@ -0,0 +1,302 @@
+		port = "8080"
+	}
+
+	log.Printf("Listening on port %s", port)
+	if err := http.ListenAndServe(":"+port, nil); err != nil {
+		log.Fatal(err)
+	}
+
+}
+
+// indexHandler handles requests to the / route.
+func indexHandler(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "GET":
+		if err := showTotals(w, r); err != nil {
+			log.Printf("showTotals: %v", err)
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+		}
+	case "POST":
+		if err := saveVote(w, r); err != nil {
+			log.Printf("saveVote: %v", err)
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+		}
+	default:
+		http.Error(w, fmt.Sprintf("HTTP Method %s Not Allowed", r.Method), http.StatusMethodNotAllowed)
+	}
+}
+
+// recentVotes returns a slice of the last 5 votes cast.
+func recentVotes() ([]vote, error) {
+	var votes []vote
+	rows, err := db.Query(`SELECT TOP 5 candidate, time_cast FROM votes ORDER BY time_cast DESC`)
+	if err != nil {
+		return votes, fmt.Errorf("DB.Query: %v", err)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		nextVote := vote{}
+		err := rows.Scan(&nextVote.Candidate, &nextVote.VoteTime)
+		if err != nil {
+			return votes, fmt.Errorf("Rows.Scan: %v", err)
+		}
+		votes = append(votes, nextVote)
+	}
+	return votes, nil
+}
+
+// currentTotals returns a templateData structure for populating the web page.
+func currentTotals() (templateData, error) {
+
+	// get total votes for each candidate
+	var tabVotes, spaceVotes uint
+	err := db.QueryRow(`SELECT count(vote_id) FROM votes WHERE candidate='TABS'`).Scan(&tabVotes)
+	if err != nil {
+		return templateData{}, fmt.Errorf("DB.QueryRow: %v", err)
+	}
+	err = db.QueryRow(`SELECT count(vote_id) FROM votes WHERE candidate='SPACES'`).Scan(&spaceVotes)
+	if err != nil {
+		return templateData{}, fmt.Errorf("DB.QueryRow: %v", err)
+	}
+
+	var voteDiffStr string = voteDiff(int(math.Abs(float64(tabVotes) - float64(spaceVotes)))).String()
+
+	latestVotesCast, err := recentVotes()
+	if err != nil {
+		return templateData{}, fmt.Errorf("recentVotes: %v", err)
+	}
+	return templateData{tabVotes, spaceVotes, voteDiffStr, latestVotesCast}, nil
+
+}
+
+// showTotals renders an HTML template showing the current vote totals.
+func showTotals(w http.ResponseWriter, r *http.Request) error {
+
+	totals, err := currentTotals()
+	if err != nil {
+		return fmt.Errorf("currentTotals: %v", err)
+	}
+	err = parsedTemplate.Execute(w, totals)
+	if err != nil {
+		return fmt.Errorf("Template.Execute: %v", err)
+	}
+	return nil
+}
+
+// saveVote saves a vote passed as http.Request form data.
+func saveVote(w http.ResponseWriter, r *http.Request) error {
+	if err := r.ParseForm(); err != nil {
+		return fmt.Errorf("Request.ParseForm: %v", err)
+	}
+
+	var team string
+	if teamprop, ok := r.Form["team"]; ok {
+		team = teamprop[0]
+	} else {
+		return fmt.Errorf("team property missing from form submission")
+	}
+
+	// [START cloud_sql_sqlserver_databasesql_connection]
```

Comment:
Normally, samples include the required imports, as it can cause confusion for users when they aren't sure what to import.

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/CONTRIBUTING.md#include-imports-and-flags-in-region-tags

For some tutorials it makes sense to have a separate import block and include it at the top of the page, not in every snippet.

We also try to include the function declaration. Otherwise, if users copy the code into a file (or look at it embedded on cloud.google.com), it doesn't compile.

Is there any way we can follow those patterns here?

Reply:
Normally I would agree with the preference, but in the original design for these samples we opted not to include imports since they were part of larger app (which was important because we wanted to emphasize connection pooling in a "common use case"). Specifically, the reasoning was:
1. We might either include unnecessary or unused dependencies, which wouldn't be caught be testing
2. User's copy/pasting was less likely for these samples, since most connections to a database are done in the context of larger application

This is consistent with the [MySQL Go sample, as well as the samples for the other languages](https://cloud.google.com/sql/docs/mysql/manage-connections#go).